src/core/server.cc

```diff
@@ -340,13 +340,19 @@ InferenceServer::ModelIsReady(
 
   ScopedAtomicIncrement inflight(inflight_request_counter_);
 
-  std::shared_ptr<InferenceBackend> backend;
-  RETURN_IF_ERROR(GetInferenceBackend(model_name, model_version, &backend));
-
-  ModelReadyState state;
-  RETURN_IF_ERROR(model_repository_manager_->GetModelState(
-      model_name, backend->Version(), &state));
-  *ready = (state == ModelReadyState::MODEL_READY);
+  std::map<int64_t, ModelReadyState> states;
```

Comment:
backend->Version() gives us the actual version, doesn't it? So we don't have to worry about -1. I think the original code is correct.

Reply:
`GetInferenceBackend()` returns error if the backend is unloaded / unavailable. I guess in that case we can just return success and set ready to false. Then we can use the original code.

Reply:
Sounds good.